crates/lib3h/src/transport/protocol.rs

```diff
@@ -50,6 +52,78 @@ pub type DynTransportActor = Box<
     >,
 >;
 
+/// this is awkward... I *think* it'll go away with ghost-actor proc-macro
+pub struct TransportEndpointAsActor {
+    endpoint_parent: Option<TransportActorParentEndpoint>,
+    endpoint_self: Detach<
+        GhostContextEndpoint<
+            Self,
+            RequestToParent,
+            RequestToParentResponse,
+            RequestToChild,
+            RequestToChildResponse,
+            TransportError,
+        >,
+    >,
+    endpoint_inner: Detach<TransportActorParentContextEndpoint<Self>>,
+}
+
+impl TransportEndpointAsActor {
+    pub fn new(inner: TransportActorParentEndpoint) -> Self {
+        let (endpoint_parent, endpoint_self) = create_ghost_channel();
+        let endpoint_parent = Some(endpoint_parent);
+        let endpoint_self = Detach::new(
+            endpoint_self
+                .as_context_endpoint_builder()
+                .request_id_prefix("enc_to_parent_")
+                .build(),
+        );
+        Self {
+            endpoint_parent,
+            endpoint_self,
+            endpoint_inner: Detach::new(inner.as_context_endpoint_builder().build::<Self>()),
+        }
+    }
+}
+
+impl
+    GhostActor<
+        RequestToParent,
+        RequestToParentResponse,
+        RequestToChild,
+        RequestToChildResponse,
+        TransportError,
+    > for TransportEndpointAsActor
+{
+    fn take_parent_endpoint(&mut self) -> Option<TransportActorParentEndpoint> {
+        std::mem::replace(&mut self.endpoint_parent, None)
+    }
+
+    fn process_concrete(&mut self) -> GhostResult<WorkWasDone> {
+        detach_run!(&mut self.endpoint_self, |es| es.process(self))?;
+        for mut msg in self.endpoint_self.as_mut().drain_messages() {
+            let data = msg.take_message().expect("exists");
+            self.endpoint_inner.request(
+                Lib3hSpan::todo(),
+                data,
+                Box::new(move |_, response| {
+                    msg.respond(match response {
+                        GhostCallbackData::Timeout => Err("timeout".into()),
+                        GhostCallbackData::Response(r) => r,
+                    })?;
+                    Ok(())
+                }),
+            )?;
+        }
+        detach_run!(&mut self.endpoint_inner, |it| it.process(self))?;
+        for mut msg in self.endpoint_inner.as_mut().drain_messages() {
+            let data = msg.take_message().expect("exists");
+            self.endpoint_self.publish(Lib3hSpan::todo(), data)?;
+        }
+        Ok(false.into())
```

Comment:
So, Is this function always tells us that the work was never done ? :thinking:

Reply:
hehe, yeah, just a stub until @struktured gives us a pattern to follow : )